### PR TITLE
Use legacy number series for MHR number generation.

### PR DESCRIPTION
**Pull request by doug-lovett**
*Issue #:* /bcgov/entity#16674

*Description of changes:*
- Change MHR number generation to use the same strategy as the legacy application.
- Refactor large DB2 utils file to move queries to a separate file.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

**Review by cameron-eyds — APPROVED** [review]
Looks good!

**Review by dimak1 — APPROVED** [review]

File: mhr_api/tests/unit/models/db2/test_db2_utils.py [hunk 1]
```diff
@@ -76,7 +76,7 @@
 
 # testdata pattern is ({account_id}, {collapse}, {start_value}, {end_value}, {second_filter_name},
 #                      {second_filter_value}, {mhr_numbers}, {expected_date_clause}, {expected_second_clause})
-TEST_QUERY_FILTER_DATA_MULTIPLE = [
```
Comment (orelbn) [inline mhr_api/tests/unit/models/db2/test_db2_utils.py:79]: Did you mean to remove the space here?

**Review by orelbn — APPROVED** [review]
One super tiny comment

File: mhr_api/tests/unit/models/db2/test_db2_utils.py (hunk 1 above)
Comment (doug-lovett) [reply mhr_api/tests/unit/models/db2/test_db2_utils.py:79]: Thanks, I'll put it back in the next PR: I closed this one while you were reviewing it and the change is now in DEV.

**Pull request merged**